middleware/logging_test.go

```diff
@@ -96,6 +97,64 @@ func TestLoggingRequestsAtInfoLevel(t *testing.T) {
 	}
 }
 
+func TestLoggingRequestWithExcludedHeaders(t *testing.T) {
+	for _, tc := range []struct {
+		name              string
+		setHeaderList     []string
+		excludeHeaderList []string
+		mustNotContain    []string
+	}{
+		{
+			name:           "Default excluded headers are excluded",
+			setHeaderList:  defaultExcludedHeaders,
+			mustNotContain: defaultExcludedHeaders,
+		},
+		{
+			name:              "Extra configured header is also excluded",
+			setHeaderList:     append(defaultExcludedHeaders, "X-Secret-Header"),
+			excludeHeaderList: []string{"X-Secret-Header"},
+			mustNotContain:    append(defaultExcludedHeaders, "X-Secret-Header"),
+		},
+		{
+			name:              "Multiple extra configured headers are also excluded",
+			setHeaderList:     append(defaultExcludedHeaders, "X-Secret-Header", "X-Secret-Header-2"),
+			excludeHeaderList: []string{"X-Secret-Header", "X-Secret-Header-2"},
+			mustNotContain:    append(defaultExcludedHeaders, "X-Secret-Header", "X-Secret-Header-2"),
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			buf := bytes.NewBuffer(nil)
+			logrusLogger := logrus.New()
+			logrusLogger.Out = buf
+			logrusLogger.Level = logrus.DebugLevel
+
+			loggingMiddleware := NewLogMiddleware(logging.Logrus(logrusLogger), true, false, nil, tc.excludeHeaderList)
+
+			handler := func(w http.ResponseWriter, r *http.Request) {
+				_, _ = io.WriteString(w, "<html><body>Hello world!</body></html>")
+			}
+			loggingHandler := loggingMiddleware.Wrap(http.HandlerFunc(handler))
+
+			req := httptest.NewRequest("GET", "http://example.com/foo", nil)
+			for _, header := range tc.setHeaderList {
+				req.Header.Set(header, header)
+			}
+			recorder := httptest.NewRecorder()
+
+			w := errorWriter{
+				err: nil,
+				w:   recorder,
+			}
+			loggingHandler.ServeHTTP(w, req)
+
+			body := buf.String()
```

Comment:
"body" seems a confusing choice of name for this, which is the log output, where we are dealing with http requests which have a member `Body`.